src/framework/auth.rs

```diff
@@ -13,10 +15,14 @@ pub trait AuthClient {
 impl AuthClient for RequestBuilder {
     fn auth(self, credentials: &Credentials) -> RequestBuilder {
         match credentials {
-            Credentials::User { key, email } => self
-                .header("X-Auth-Key", key.as_str())
-                .header("X-Auth-Email", email.as_str()),
+            Credentials::UserAuthKey { email, key } => self
+                .header("X-Auth-Email", email.as_str())
+                .header("X-Auth-Key", key.clone()),
+            Credentials::UserAuthToken { email, token } => self
+                .header("X-Auth-Email", email.as_str())
```

Comment:
according to the API docs, tokens are used by themselves, not in tandem with email